django/db/models/aggregates.py

```diff
@@ -196,6 +245,52 @@ def _get_repr_options(self):
         return {**super()._get_repr_options(), "sample": self.function == "STDDEV_SAMP"}
 
 
+class StringAgg(Aggregate):
+    template = "%(function)s(%(distinct)s%(expressions)s%(order_by)s)%(filter)s"
+    function = "STRING_AGG"
+    name = "StringAgg"
+    allow_distinct = True
+    allow_order_by = True
+    output_field = TextField()
+
+    def __init__(self, expression, delimiter, **extra):
+        delimiter = Value(str(delimiter))
+        super().__init__(expression, delimiter, **extra)
+
+    def as_sql(self, compiler, connection, **extra_context):
+        return super().as_sql(compiler, connection, **extra_context)
+
+    def as_oracle(self, compiler, connection, **extra_context):
+        if self.order_by:
+            template = "%(function)s(%(distinct)s%(expressions)s) WITHIN GROUP %(order_by)s%(filter)s"
+        else:
+            template = "%(function)s(%(distinct)s%(expressions)s)%(filter)s"
+
+        return self.as_sql(
+            compiler,
+            connection,
+            function="LISTAGG",
+            template=template,
+            **extra_context,
+        )
+
+    def as_mysql(self, compiler, connection, **extra_context):
+        return self.as_sql(
+            compiler,
+            connection,
+            function="GROUP_CONCAT",
+            **extra_context,
+        )
+
+    def as_sqlite(self, compiler, connection, **extra_context):
+        return self.as_sql(
+            compiler,
+            connection,
+            function="GROUP_CONCAT",
```

Comment:
I _think_ [we can use `STRING_AGG` on SQLite as well](https://www.sqlite.org/lang_aggfunc.html#group_concat)?

Reply:
I expected that as well, but I was getting errors that the function wasn't defined. It turns out, `STRING_AGG` was added in 3.44.0 at the same time as the `order_by` was added to the behaviors. 

https://www.sqlite.org/changes.html#version_3_44_0

Since my local sqlite versions have been 3.37, it was failing. Not sure if we have a preference on which to use for sqlite in this case. `GROUP_CONCAT` feels like it is a bit more flexible since it is older, but I don't know if the behaviors are exactly the same or not under the hood. I'll have to figure that out once I get a newer version of sqlite hooked up to my tests.

Reply:
I missed that, let's stick to `GROUP_CONCAT` then. I guess we could add a `connection.get_database_version() < (3, 44)` check so eventually when we drop support for < 3.44 we'll drop that code at the same time.